tests/test_sidebar.c

```diff
@@ -137,12 +137,16 @@ void test_sidebar_openfiles_tree(void)
 
 int main(int argc, char **argv)
 {
-	g_test_init(&argc, &argv, NULL);
 	/* Not sure if we can really continue without DISPLAY. Fake X display perhaps?
 	 *
 	 * This test seems to work, at least.
 	 */
-	gtk_init_check(&argc, &argv);
+	if(!gtk_init_check(&argc, &argv)) {
+		printf("GTK initialisation failed; skipping. Running inside a headless environment?\n");
```

Comment:
Please adapt the above comment to the new code. I'm also unsure if printf output is really desirable during test runs.